### PR TITLE
Fix emscripten_webgl_make_context_current under wasm64/CAN_ADDRESS_2GB

### PR DESCRIPTION
**Pull request by sbc100**
The first problem here was that
emscripten_webgl_make_context_current_calling_thread had the
wrong signature.
    
The second issue was use of `>> 2` on a pointer in
`emscripten_webgl_do_create_context` which doesn't work under wasm64
or CAN_ADDRESS_2GB.

### CODE REVIEW

**Comment by sbc100** [comment]
Supersedes #21187

**Comment by dschuff** [comment]
the fix to emscripten_webgl_do_create_context is separate from the fix to emscripten_webgl_make_context_current, right? Can you add a commit message that describes what's actually changing?

File: test/test_browser.py [hunk 1]
```diff
@@ -4987,10 +4987,14 @@ def test_pthread_run_script(self, args):
     'manual_css': (['-sPROXY_TO_PTHREAD', '-pthread', '-sOFFSCREEN_FRAMEBUFFER', '-DTEST_EXPLICIT_CONTEXT_SWAP=1', '-DTEST_MANUALLY_SET_ELEMENT_CSS_SIZE=1'], False),
   })
   def test_emscripten_animate_canvas_element_size(self, args, main_loop):
-    cmd = ['-lGL', '-O3', '-g2', '--shell-file', test_file('canvas_animate_resize_shell.html'), '-sGL_DEBUG', '--threadprofiler', '-sASSERTIONS'] + args
+    cmd = ['-lGL', '-O3', '-g2', '--shell-file', test_file('canvas_animate_resize_shell.html'), '-sGL_DEBUG', '-sASSERTIONS'] + args
+    if not self.is_2gb() and not self.is_4gb():
+      # Thread profiles does not yet work with large pointers.
```
Comment (kripken) [inline test/test_browser.py:4992]:
```suggestion
      # Thread profiler does not yet work with large pointers.
```

**Review by kripken — APPROVED** [review]

**Comment by sbc100** [comment]
> the fix to emscripten_webgl_do_create_context is separate from the fix to emscripten_webgl_make_context_current, right? Can you add a commit message that describes what's actually changing?

Done

**Pull request merged**